controllers/windowsmachine_controller.go

```diff
@@ -247,6 +247,12 @@ func (r *WindowsMachineReconciler) Reconcile(ctx context.Context, request ctrl.R
 		err := r.client.Get(ctx, kubeTypes.NamespacedName{Namespace: machine.Status.NodeRef.Namespace,
 			Name: machine.Status.NodeRef.Name}, node)
 		if err != nil {
+			// Delete a configured/running machine if no associated node can be determined (node object deleted)
+			if k8sapierrors.IsNotFound(err) {
+				log.Info("machine has no associated node", "name", machine.GetName())
+				requeueNeeded, err := r.deleteMachineIfAllowed(machine)
+				return ctrl.Result{Requeue: requeueNeeded}, err
```

Comment:
If the machine deletion is not allowed because it falls below the minHealthyCount, we will not requeue and return the error here. How do we re-mediate that?
Did we consider not re-queuing?

Reply:
Thanks for the explanation and that is the behavior I was expecting. I must have misread the function and thought we are returning error.